languages/common-lisp/exercises/concept/lillys-lasagna-leftovers/.docs/introduction.md

```diff
@@ -0,0 +1,56 @@
+In Common Lisp a functions argument list (also known as a ('lambda list')[lambda-list]) can have arguments of different types. These different types are designated with the use of ("lambda list keywords")[lambda-list-keyword] which all begin with `&`. The most commonly used types are optional, keyword and rest arguments types. Every parameter in the lambda list after a particular lambda list keyword is will be of that type. A lambda list keyword can only be used once in a lambda list.
+
+## default-parameters
+
+In Common Lisp a function can have some arguments are are optional. These are designated in the lambda list by `&optional` lambda list keyword. A parameter will be bound to the value `nil` if it is not specified. If there are several optional parameters they are bound in order. Default values can be specified for optional parameters. Finally a symbol an be specified for each optional parameter which will be bound to true or false depending on whether that parameter was supplied by the caller of the function (this is referred to as the "supplied-p parameter").
+
+```lisp
+(defun default-parameters (&optional x (y 'default) (z nil z-supplied-p))
+  (list x y (if z-supplied-p (list :z-was-supplied z)
+                             (list :z-was-not-supplied z))))
+
+(default-parameters) ; => '(nil 'default (:z-was-not-supplied nil))
```

Comment:
I've been making my "returns" (the bit right of the `=>`) all caps and without quotes (exactly how it comes back from lisp, like: `(NIL DEFAULT (:Z-WAS-NOT-SUPPLIED NIL))`, but I suppose that's not really all that important. Up to you, which you prefer!

Reply:
Ah good point - I'll make a pass through all the files i created in this exercise and try to make it consistent.